### ISSUE TITLE
Discrepency between the training results and given checkinpoint

### ISSUE THREAD

**Issue by Moonsofang** [csguoh/MambaIR#132]
Hello author, thank you for the open-source code! I'm a beginner in the field of super-resolution and encountered some problems while trying to reproduce the results of MambaIRv2_lightSR in the paper. This is my config setting and training log. After 500,000 training iterations, I did not get the results described in the paper, and the performance on the Urban test set differed significantly. I would like to ask if there are any problems with the training? And were any further optimizations performed on the model after training? 

```
Version Information: 
	BasicSR: 1.3.5
	PyTorch: 2.0.1
	TorchVision: 0.15.2
2025-12-14 21:53:54,971 basicsr/train.py INFO: 
  name: MambaIRv2_lightSR_x2
  model_type: MambaIRv2LightModel
  scale: 2
  num_gpu: 2
  manual_seed: 10
  datasets:[
    train:[
      task: SR
      name: DF2K
      type: PairedImageDataset
      dataroot_gt: ['./train_dataset/DIV2K/HR']
      dataroot_lq: ['./train_dataset/DIV2K/X2']
      filename_tmpl: {}x2
      io_backend:[
        type: disk
      ]
      gt_size: 128
      use_hflip: True
      use_rot: True
      use_shuffle: True
      num_worker_per_gpu: 8
      batch_size_per_gpu: 16
      dataset_enlarge_ratio: 100
      prefetch_mode: None
      phase: train
      scale: 2
    ]
    val:[
      name: Set14
      type: PairedImageDataset
      dataroot_gt: ./test_dataset/SR/Set14/HR/
      dataroot_lq: ./test_dataset/SR/Set14/LR_bicubic/X2/
      filename_tmpl: {}x2
      io_backend:[
        type: disk
      ]
      phase: val
      scale: 2
    ]
  ]
  network_g:[
    type: MambaIRv2Light
    upscale: 2
    in_chans: 3
    img_size: 64
    img_range: 1.0
    embed_dim: 48
    d_state: 8
    depths: [5, 5, 5, 5]
    num_heads: [4, 4, 4, 4]
    window_size: 16
    inner_rank: 32
    num_tokens: 64
    convffn_kernel_size: 5
    mlp_ratio: 1.0
    upsampler: pixelshuffledirect
    resi_connection: 1conv
  ]
  path:[
    experiments_root: ./experiments/train_result/MambaIRv2_lightSR_x2
    pretrain_network_g: None
    strict_load_g: True
    resume_state: None
    models: ./MambaIR/experiments/train_result/MambaIRv2_lightSR_x2/models
    training_states: ./MambaIR/experiments/train_result/MambaIRv2_lightSR_x2/training_states
    log: ./MambaIR/experiments/train_result/MambaIRv2_lightSR_x2
    visualization: ./MambaIR/experiments/train_result/MambaIRv2_lightSR_x2/visualization
  ]
  train:[
    optim_g:[
      type: Adam
      lr: 0.0002
      weight_decay: 0
      betas: [0.9, 0.99]
    ]
    scheduler:[
      type: MultiStepLR
      milestones: [250000, 400000, 450000, 475000]
      gamma: 0.5
    ]
    total_iter: 500000
    warmup_iter: -1
    pixel_opt:[
      type: L1Loss
      loss_weight: 1.0
      reduction: mean
    ]
  ]
  val:[
    val_freq: 5000.0
    save_img: False
    metrics:[
      psnr:[
        type: calculate_psnr
        crop_border: 2
        test_y_channel: True
      ]
    ]
  ]
  logger:[
    print_freq: 200
    save_checkpoint_freq: 5000.0
    use_tb_logger: True
    wandb:[
      project: None
      resume_id: None
    ]
  ]
  dist_params:[
    backend: nccl
    port: 29500
  ]
  dist: True
  rank: 0
  world_size: 2
  auto_resume: False
  is_train: True
  root_path: ./MambaIR
```

Part of the training log is as follows (Different from the 34.09 PSNR on Set14 in the paper):
```
2025-12-17 23:07:43,378  INFO: [Mamba..][epoch:198, iter: 495,200, lr:(1.250e-05,)] [eta: 0:42:33, time (data): 0.499 (0.004)] l_pix: 1.1147e-02 
2025-12-17 23:09:22,427  INFO: [Mamba..][epoch:198, iter: 495,400, lr:(1.250e-05,)] [eta: 0:40:47, time (data): 0.495 (0.003)] l_pix: 9.0709e-03 
2025-12-17 23:11:01,145  INFO: [Mamba..][epoch:198, iter: 495,600, lr:(1.250e-05,)] [eta: 0:39:00, time (data): 0.493 (0.003)] l_pix: 1.4908e-02 
2025-12-17 23:12:40,106  INFO: [Mamba..][epoch:198, iter: 495,800, lr:(1.250e-05,)] [eta: 0:37:14, time (data): 0.495 (0.003)] l_pix: 1.2386e-02 
2025-12-17 23:14:18,823  INFO: [Mamba..][epoch:198, iter: 496,000, lr:(1.250e-05,)] [eta: 0:35:27, time (data): 0.494 (0.003)] l_pix: 1.4057e-02 
2025-12-17 23:15:57,366  INFO: [Mamba..][epoch:198, iter: 496,200, lr:(1.250e-05,)] [eta: 0:33:41, time (data): 0.493 (0.003)] l_pix: 1.2842e-02 
2025-12-17 23:17:35,641  INFO: [Mamba..][epoch:198, iter: 496,400, lr:(1.250e-05,)] [eta: 0:31:54, time (data): 0.491 (0.003)] l_pix: 1.1097e-02 
2025-12-17 23:19:14,628  INFO: [Mamba..][epoch:198, iter: 496,600, lr:(1.250e-05,)] [eta: 0:30:08, time (data): 0.496 (0.003)] l_pix: 1.2828e-02 
2025-12-17 23:20:53,624  INFO: [Mamba..][epoch:198, iter: 496,800, lr:(1.250e-05,)] [eta: 0:28:21, time (data): 0.495 (0.004)] l_pix: 1.3524e-02 
2025-12-17 23:22:33,246  INFO: [Mamba..][epoch:198, iter: 497,000, lr:(1.250e-05,)] [eta: 0:26:35, time (data): 0.498 (0.003)] l_pix: 1.2749e-02 
2025-12-17 23:24:12,229  INFO: [Mamba..][epoch:198, iter: 497,200, lr:(1.250e-05,)] [eta: 0:24:49, time (data): 0.496 (0.003)] l_pix: 1.2934e-02 
2025-12-17 23:25:51,518  INFO: [Mamba..][epoch:198, iter: 497,400, lr:(1.250e-05,)] [eta: 0:23:02, time (data): 0.496 (0.003)] l_pix: 1.3998e-02 
2025-12-17 23:28:20,947  INFO: [Mamba..][epoch:199, iter: 497,600, lr:(1.250e-05,)] [eta: 0:21:16, time (data): 0.900 (0.406)] l_pix: 1.0848e-02 
2025-12-17 23:30:00,543  INFO: [Mamba..][epoch:199, iter: 497,800, lr:(1.250e-05,)] [eta: 0:19:30, time (data): 0.497 (0.004)] l_pix: 1.1353e-02 
2025-12-17 23:31:40,391  INFO: [Mamba..][epoch:199, iter: 498,000, lr:(1.250e-05,)] [eta: 0:17:43, time (data): 0.499 (0.004)] l_pix: 9.4114e-03 
2025-12-17 23:33:19,808  INFO: [Mamba..][epoch:199, iter: 498,200, lr:(1.250e-05,)] [eta: 0:15:57, time (data): 0.497 (0.003)] l_pix: 1.4536e-02 
2025-12-17 23:34:59,216  INFO: [Mamba..][epoch:199, iter: 498,400, lr:(1.250e-05,)] [eta: 0:14:10, time (data): 0.497 (0.003)] l_pix: 1.0980e-02 
2025-12-17 23:36:38,015  INFO: [Mamba..][epoch:199, iter: 498,600, lr:(1.250e-05,)] [eta: 0:12:24, time (data): 0.493 (0.003)] l_pix: 1.3262e-02 
2025-12-17 23:38:17,168  INFO: [Mamba..][epoch:199, iter: 498,800, lr:(1.250e-05,)] [eta: 0:10:37, time (data): 0.495 (0.003)] l_pix: 9.6415e-03 
2025-12-17 23:39:56,170  INFO: [Mamba..][epoch:199, iter: 499,000, lr:(1.250e-05,)] [eta: 0:08:51, time (data): 0.497 (0.003)] l_pix: 1.3814e-02 
2025-12-17 23:41:35,144  INFO: [Mamba..][epoch:199, iter: 499,200, lr:(1.250e-05,)] [eta: 0:07:05, time (data): 0.495 (0.003)] l_pix: 1.0077e-02 
2025-12-17 23:43:13,998  INFO: [Mamba..][epoch:199, iter: 499,400, lr:(1.250e-05,)] [eta: 0:05:18, time (data): 0.494 (0.003)] l_pix: 1.1685e-02 
2025-12-17 23:44:53,189  INFO: [Mamba..][epoch:199, iter: 499,600, lr:(1.250e-05,)] [eta: 0:03:32, time (data): 0.494 (0.003)] l_pix: 1.4009e-02 
2025-12-17 23:46:32,171  INFO: [Mamba..][epoch:199, iter: 499,800, lr:(1.250e-05,)] [eta: 0:01:45, time (data): 0.493 (0.003)] l_pix: 1.5988e-02 
2025-12-17 23:48:11,492  INFO: [Mamba..][epoch:199, iter: 500,000, lr:(1.250e-05,)] [eta: 0:00:00, time (data): 0.495 (0.003)] l_pix: 1.0976e-02 
2025-12-17 23:48:11,493 basicsr/train.py INFO: Saving models and training states.
2025-12-17 23:48:18,930 ./MambaIR/basicsr/models/sr_model.py INFO: Validation Set14
	 # psnr: 33.9968	Best: 33.9972 @ 445000 iter

2025-12-17 23:49:06,725 basicsr/train.py INFO: End of training. Time consumed: 3 days, 1:54:22
2025-12-17 23:49:06,726 basicsr/train.py INFO: Save the latest model.
2025-12-17 23:49:13,372 ./MambaIR/basicsr/models/sr_model.py INFO: Validation Set14
	 # psnr: 33.9872	Best: 33.9972 @ 445000 iter
``` 

Would it be convenient to provide the training logs for the MambaIRv2 lightsr?

**Comment by csguoh** [comment]
Hi, you may refer to [this log file](https://drive.google.com/file/d/1yeZ7PTAQfoAoQhkBaCWSkPPtAwxyG9bO/view?usp=sharing). Since it has been a long time ago, I am not sure whether this config is the default one. But it seems the result matches the one in our paper.
